lib/CUEParser/src/CUEParser.cpp

```diff
@@ -104,12 +104,11 @@ const CUETrackInfo *CUEParser::next_track()
 
             if (index == 0)
             {
-                m_track_info.pregap_start = time;
                 m_track_info.track_start = time;
+                got_paused = true;
```

Comment:
Definitely, that change is vague and I should have left more comments. Hopefully the additional commit should help. For context (and to help make sure I have this right) here's my understanding of gaps with CDs and cue sheets. Please correct me if this seems wrong, I'm no expert on the subject.

On real CDs "pauses" are sections of the continuous data stream where there is control information but no corresponding data written: effectively periods of silence. These are normally marked in cue sheets using either an INDEX 00 or a PREGAP command. INDEX 00 has matching data in the .bin, usually silence (though apparently it can contain audio samples as well), whereas PREGAP has _no_ matching data in the file and the silence needs to be inserted by the playback software. This does not include the mandatory 2 second pause at the start of a disk, which is assumed in cue sheets and is omitted.

The `got_paused` variable was added to resolve situations where the first track of a disk has an INDEX 00 entry, which was the case on a test disk I was using that has hidden track one audio (Rammstein's _Reise Reise_, though there are plenty of others out there that use this trick). In the old code flow this was causing the pregap timestamp to be set to 0, which was then being ignored due to it being == 0 in the later check.

Overall the new changes try to sharpen the distinction between 'track start', index 0 or index 1, and 'data start', where real samples begin at index 1.

All that said, there is still a problem with the PREGAP command implementation. Like a CD player, the audio code assumes samples are continuous, and it has no way to know that a missing amount of silence should be re-inserted into playback. I don't have a good answer about how to address this apart from blithely saying 'use software that inserts actual silence,' which thankfully seems to be the norm for most archiving software I've experienced. A subsequent PR might be needed to figure this out.

Sorry for the long answer. Let me know if I'm mistaken about anything or if more clarification should be inserted in the code.